wormhole-connect/src/utils/routes/cctpManual.ts

```diff
@@ -0,0 +1,612 @@
+    // only works on EVM
+    const chainContext = wh.getContext(
+      sendingChain,
+    ) as EthContext<WormholeContext>;
+    const tokenMessenger =
+      wh.mustGetContracts(sendingChain).cctpContracts?.cctpTokenMessenger;
+    const circleTokenMessenger = await TokenMessenger__factory.connect(
+      tokenMessenger!,
+      wh.getSigner(fromChainId)!,
+    );
+    const tokenAddr = await wh.mustGetForeignAsset(
+      token as TokenId,
+      sendingChain,
+    );
+
+    // approve
+    await chainContext.approve(
+      sendingChain,
+      circleTokenMessenger.address,
+      tokenAddr,
+      parsedAmt,
+    );
+
+    console.log('About to send 2');
+    const tx = await circleTokenMessenger.populateTransaction.depositForBurn(
+      parsedAmt,
+      getDomainCCTP(wh.toChainId(recipientChain)),
+      chainContext.context.formatAddress(recipientAddress, recipientChain),
+      chainContext.context.parseAddress(tokenAddr, sendingChain),
+    );
+
+    const sentTx = await wh.getSigner(fromChainName)?.sendTransaction(tx);
+    const rx = await sentTx?.wait();
+    if (!rx) throw "Transaction didn't go through";
+    const txId = await signAndSendTransaction(
+      fromChainName,
+      rx,
+      TransferWallet.SENDING,
+    );
+    console.log('signed and sent');
+    wh.registerProviders();
+    return txId;
+  }
+
+  async redeem(
+    destChain: ChainName | ChainId,
+    messageInfo: CCTPInfo,
+    payer: string,
+  ): Promise<string> {
+    const context: any = wh.getContext(destChain);
+    const circleMessageTransmitter =
+      context.contracts.mustGetContracts(destChain).cctpContracts
+        ?.cctpMessageTransmitter;
+    const connection = wh.mustGetSigner(destChain);
+    const contract = MessageTransmitter__factory.connect(
+      circleMessageTransmitter,
+      connection,
+    );
+    const tx = await contract.receiveMessage(
+      messageInfo.message,
+      messageInfo.signedAttestation,
+    );
+    await tx.wait();
+    return tx.hash;
+  }
+
+  async parseMessage(
+    messageInfo: CCTPInfo,
+  ): Promise<ParsedMessage | ParsedRelayerMessage> {
+    console.log('here is the message to parse');
+    console.log(JSON.stringify(messageInfo));
+    const tokenId: TokenId = {
+      chain: messageInfo.fromChain,
+      address: messageInfo.burnToken,
+    };
+    const token = getTokenById(tokenId);
+    const decimals = await wh.fetchTokenDecimals(
+      tokenId,
+      messageInfo.fromChain,
+    );
+    let relayerPart = undefined;
+    if (messageInfo.relayerPayloadId !== undefined) {
+      relayerPart = {
+        relayerPayloadId: messageInfo.relayerPayloadId,
+        to: messageInfo.mintRecipient,
+        relayerFee: messageInfo.relayerFee,
+        toNativeTokenAmount: messageInfo.toNativeTokenAmount,
+      };
+    }
+    return {
+      sendTx: messageInfo.transactionHash,
+      sender: messageInfo.depositor,
+      amount: messageInfo.amount.toString(),
+      payloadID: PayloadType.MANUAL,
```

Comment:
I'm not sure this is correct.  payloadID is the first byte which identifies the type of transfer. Maybe we can add another field for Automatic vs Manual